pkgs/top-level/python-packages.nix

```diff
@@ -20085,6 +20085,36 @@ in modules // {
     };
   };
 
+  Theano = buildPythonPackage rec {
+    name = "Theano-0.8.1";
+
+    disabled = isPyPy || pythonOlder "2.6" || (isPy3k && pythonOlder "3.3");
+
+    src = pkgs.fetchurl {
+      url = "https://pypi.python.org/packages/source/T/Theano/${name}.tar.gz";
+      sha256 = "17dikk94r8bzkxg976srqlhj5c7phs9gl837iabyfdpixkbrl79g";
+    };
+
+    #preCheck = ''
+    #  mkdir -p check-phase
+    #  export HOME=$(pwd)/check-phase
+    #'';
+    doCheck = false;
+    # takes far too long, also throws "TypeError: sort() missing 1 required positional argument: 'a'"
+    # when run from the installer, and testing with Python 3.5 hits github.com/Theano/Theano/issues/4276,
+    # the fix for which hasn't been merged yet.
+
+    # keep Nose around since running the tests by hand is possible from Python or bash
+    propagatedBuildInputs = [ stdenv pkgs.blas ] ++ (with self; [ nose numpy pydot_ng scipy six ]);
```

Comment:
I haven't tested without this, but I suspect it's necessary since compiling Theano itself is relatively trivial, but Theano itself compiles code at 'runtime'.